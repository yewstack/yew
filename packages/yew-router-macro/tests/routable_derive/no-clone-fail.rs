#![no_implicit_prelude]

#[derive(::yew_router::Routable)]
enum Routes {
    #[at("/")]
    One,
    #[at("/two/:id")]
    Two {
        id: u32,
        #[bind(query)]
        query: u32,
    },
    #[at("/404")]
    #[bind(not_found)]
    NotFound,
}

fn main() {}
