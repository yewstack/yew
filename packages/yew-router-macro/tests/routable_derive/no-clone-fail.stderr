error[E0277]: the trait bound `Routes: Clone` is not satisfied
  --> $DIR/no-clone-fail.rs:3:10
   |
3  | #[derive(::yew_router::Routable)]
   |          ^^^^^^^^^^^^^^^^^^^^^^ the trait `Clone` is not implemented for `Routes`
   |
  ::: $WORKSPACE/packages/yew-router/src/routable.rs
   |
   | pub trait Routable: Sized + Clone {
   |                             ----- required by this bound in `Routable`
   |
   = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
